apps/meteor/ee/server/apps/orchestrator.js

```diff
@@ -177,16 +177,13 @@ export class AppServerOrchestrator {
 		/* eslint-disable no-await-in-loop */
 		// This needs to happen sequentially to keep track of app limits
 		for (const app of apps) {
-			const canEnable = await canEnableApp(app.getStorageItem());
+			try {
+				await canEnableApp(app.getStorageItem());
 
-			if (!canEnable) {
-				this._rocketchatLogger.warn(`App "${app.getInfo().name}" can't be enabled due to CE limits.`);
-				// We need to continue as the limits are applied depending on the app installation source
-				// i.e. if one limit is hit, we can't break the loop as the following apps might still be valid
-				continue;
+				await this.getManager().loadOne(app.getID());
+			} catch (error) {
+				this._rocketchatLogger.warn(`App "${app.getInfo().name}" could not be enabled: `, error.message);
```

Comment:
Shouldn't this try catch be around the `canEnableApp` call only? If the app can be enabled but fails to load, I imagine it should throw something else

```ts
			} catch (error) {
				this._rocketchatLogger.warn(`App "${app.getInfo().name}" could not be enabled: `, error.message);
				continue;
			}
			await this.getManager().loadOne(app.getID());
```

By the way can remove that `eslint-disable no-await-in-loop` if you change the loop to be a `for await`. The warning happens because this code would not work if `apps` was as AsyncIterator instead of a simple array.

Reply:
In both cases (`canEnableApp` throws and `loadOne` throws) we should not prevent the loop from going forward and checking the next apps - so I think the try/catch should surround both calls.

> By the way can remove that `eslint-disable no-await-in-loop` if you change the loop to be a `for await`. The warning happens because this code would not work if `apps` was as AsyncIterator instead of a simple array.

I'll be changing the loop to a `for..await`, then, thanks!